packages/dev/core/src/Materials/Textures/prePassRenderTarget.ts

```diff
@@ -61,6 +62,7 @@ export class PrePassRenderTarget extends MultiRenderTarget {
         super(name, size, count, scene, options);
 
         this.renderTargetTexture = renderTargetTexture;
+        this._renderPassIds = this.renderTargetTexture ? this.renderTargetTexture.renderPassIds as number[] : [Constants.RENDERPASS_MAIN];

```

Comment:
Actually, `RENDERPASS_MAIN` is a default value only used as a fallback value.

For eg, `Scene._renderForCamera` is doing this:

```typescript
this._engine.currentRenderPassId = camera.outputRenderTarget?.renderPassId ?? camera.renderPassId ?? Constants.RENDERPASS_MAIN;
```

So, it uses the `renderPassId` of the camera `outputRenderTarget` (if any), then the `renderPassId` from the camera (if any - normally we should always have one and stop here) and uses `RENDERPASS_MAIN` as the fallback just in case.

So, in your case using `RENDERPASS_MAIN` is probably wrong and you should instead use the one from the `camera.outputRenderTarget`/`camera` in effect when rendering into this `PrePassRenderTarget`.

Reply:
oh thanks @Popov72, this code doesn't belong in this PR this is a branching error.
But your comment stands, I'll make sure to change that in the other PR